index.android.js

```diff
@@ -189,12 +193,22 @@ class RNGoogleFit {
               obj.rawSteps = this.isConfigsEmpty(options.configs) ? [] : dev.steps
               return obj
             }, this)
-          )
-        } else {
-          callback('There is no any steps data for this period', false)
+            if (typeof callback === 'function') {
+              callback(
+                false,
+                mappedRes
+              )
+            }
+            return resolve(mappedRes)
+          } else {
+            if (typeof callback === 'function') {
+              callback('There is no any steps data for this period', false)
```

Comment:
Should we return here?

Reply:
No because otherwise you are left with an unresolved promise